xtra-libp2p-offer/src/lib.rs

```diff
@@ -1,9 +1,158 @@
+use model::Position;
+use model::Price;
+use model::Timestamp;
+use model::TxFeeRate;
+use model::Usd;
+use serde::Deserialize;
+use serde::Serialize;
+use std::fmt;
+use time::Duration;
+
 pub mod maker;
 mod protocol;
 pub mod taker;
 
 pub const PROTOCOL: &str = "/itchysats/offer/1.0.0";
 
+#[derive(Clone, Serialize, Deserialize, PartialEq)]
+pub struct MakerOffers {
+    pub long: Option<Offer>,
+    pub short: Option<Offer>,
+    pub tx_fee_rate: TxFeeRate,
+    pub funding_rate_long: FundingRate,
+    pub funding_rate_short: FundingRate,
+}
+
+impl fmt::Debug for MakerOffers {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt.debug_struct("MakerOffers")
+            .field("long_order_id", &self.long.as_ref().map(|o| o.id))
+            .field("short_order_id", &self.short.as_ref().map(|o| o.id))
+            .field("tx_fee_rate", &self.tx_fee_rate)
+            .field("funding_rate_long", &self.funding_rate_long)
+            .field("funding_rate_short", &self.funding_rate_short)
+            .finish()
+    }
+}
+
+/// A concrete order created by a maker for a taker
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct Offer {
+    pub id: OfferId,
+
+    #[serde(rename = "trading_pair")]
+    pub contract_symbol: ContractSymbol,
+
+    /// The maker's position
+    ///
+    /// Since the maker is the creator of this order this always reflects the maker's position.
+    /// When we create a Cfd we change it to be the position as seen by each party, i.e. flip the
+    /// position for the taker.
+    #[serde(rename = "position")]
+    pub position_maker: Position,
+
+    pub price: Price,
+
+    pub min_quantity: Usd,
+    pub max_quantity: Usd,
+
+    /// A selection of leverages that the maker allows for the taker
+    pub leverage_choices: Vec<Leverage>,
+
+    /// The creation timestamp as set by the maker
+    #[serde(rename = "creation_timestamp")]
+    pub creation_timestamp_maker: Timestamp,
+
+    /// The duration that will be used for calculating the settlement timestamp
+    pub settlement_interval: Duration,
+
+    pub origin: Origin,
+
+    /// The id of the event to be used for price attestation
+    ///
+    /// The maker includes this into the Order based on the Oracle announcement to be used.
+    pub oracle_event_id: BitMexPriceEventId,
+
+    pub tx_fee_rate: TxFeeRate,
+    pub funding_rate: FundingRate,
+    pub opening_fee: OpeningFee,
+}
+
+impl From<MakerOffers> for model::MakerOffers {
+    fn from(maker_offers: MakerOffers) -> Self {
+        model::MakerOffers {
+            long: maker_offers.long.map(|offer| offer.into()),
+            short: maker_offers.short.map(|offer| offer.into()),
+            tx_fee_rate: maker_offers.tx_fee_rate,
+            funding_rate_long: maker_offers.funding_rate_long,
+            funding_rate_short: maker_offers.funding_rate_short,
+        }
+    }
+}
+
+#[allow(deprecated)] // allowing for backwards compatible conversion
```

Comment:
❓ Why do we need this statement? What's deprecated here?

Reply:
Because of `leverage_taker` which is deprecated.